### ISSUE TITLE
`@stylistic/declaration-block-semicolon-space-before` fix takes the semicolon into an inline comment

### ISSUE THREAD

**Issue by firefoxic** [stylelint-stylistic/stylelint-stylistic#117]
Found while fixing #100, in the same sweep as #113, #114 and the sibling defect in `declaration-bang-space-before`. The same shape as #63, #87, #88 and #89: an inline comment ends only with a line break, so whatever the fix pulls onto its line is commented out. Here it is the semicolon that ends the declaration.

### What minimal example or steps are needed to reproduce the bug?

```less
.a {
  color: red // c
  ;
}
```

And, on the other of the two paths the fix takes, a declaration carrying `!important`:

```scss
.a {
  color: red // c
    !important;
}
```

### What minimal configuration is needed to reproduce the bug?

```json
{
  "customSyntax": "postcss-less",
  "plugins": "@stylistic/stylelint-plugin",
  "rules": {
    "@stylistic/declaration-block-semicolon-space-before": "always"
  }
}
```

`never` breaks the first example just as `always` does. The second one is broken by `always` alone, and under `postcss-scss` as well as `postcss-less`.

### What did you expect to happen?

The line break in front of the semicolon is what closes the comment, so neither option can be satisfied by touching that whitespace. The fix should leave the declaration alone and let the warning stand.

### What actually happened?

```less
.a {
  color: red // c ;
}
```

and, for the declaration carrying `!important`:

```scss
.a {
  color: red // c !important ;
}
```

The semicolon is inside the comment, so the declaration ends nowhere; in the second example `!important` is taken along with it. No warning survives the run — `--fix` reports a clean pass on a file it has just broken.

The two paths are at [`lib/rules/declaration-block-semicolon-space-before/index.js:81-88`](https://github.com/stylelint-stylistic/stylelint-stylistic/blob/main/lib/rules/declaration-block-semicolon-space-before/index.js#L81-L88):

- with `!important`, `decl.raws.important` is overwritten with `` !important ``, and whatever stood between the value and the flag — the line break closing the comment among it — goes with it;
- without, the trailing whitespace run of the value is replaced by a single space, and that run is the line break.

`never` takes the same run away on the second path. On the first it is safe, since the whitespace it cuts stands behind `!important` rather than in front of it.

The first example is spared under `postcss-scss` for the time being, and only by accident: the fix is written to a copy of the value that syntax does not print, and is dropped along the way. That is #115, and repairing it hands this rule the corruption under SCSS too.

### Do you have a proposal to fix the bug?

The shape the cluster settled on: where the text the fix would cut into ends with an inline comment, report the problem and write nothing. `endsWithInlineComment` in `lib/utils/` already answers that question, and it has to be put both to the value and to `decl.raws.important` — the latter kept rather than overwritten, so that a comment or any other layout standing in front of `!important` survives the fix.
